frontend/src/contexts/authStateContext.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { useKeycloak } from '@react-keycloak/web';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import { useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+
+export interface IAuthState {
+  ready?: boolean;
+}
+
+export const AuthStateContext = React.createContext<IAuthState>({
+  ready: false,
+});
+
+export const AuthStateContextProvider = (props: { children?: any }) => {
+  const keycloak = useKeycloakWrapper();
+  const [userInfo, setUserInfo] = React.useState<any>(null);
+  const keyCloakReady: boolean = useSelector<RootState, boolean>(state => state.keyCloakReady);
```

Comment:
/nit: please rename to `keycloakReady` to remain consistent with existing naming